src/coreclr/utilcode/loaderheap.cpp

```diff
@@ -1179,7 +1184,7 @@ BOOL UnlockedLoaderHeap::UnlockedReservePages(size_t dwSizeToCommit)
         pCurBlock = pCurBlock->pNext;
 
     if (pCurBlock != NULL)
-        m_pCurBlock->pNext = pNewBlock;
+        pCurBlock->pNext = pNewBlock;
```

Comment:
This fixes a bug that I've discovered while modifying the surrounding code.

Reply:
This linked list looks overengineered. I do not see a good reason why this maintains both `m_pCurBlock` and `m_pFirstBlock`. I think a single pointer that points to the head would be enough.